### ISSUE TITLE
Upgrade to Confluent Plattform 6.0

### ISSUE THREAD

**Issue by ueisele** [NovatecConsulting/showcase-kafka-iot-emob#6]
At the moment we are using Confluent Plattform 5.5.1. We should upgrade our Docker images and Java libraries to Confluent Plattform 6.0.
One advantage of this would be, that we can use the KsqlDB version which is part of CP 6.0. At the moment we have to use KsqlDB 0.11.0, because it supports some required features, which have not been supported by Ksql of CP 5.5.
Because it is a showcase, we should try to keep up to date with the current available Confluent Plattform as long we actively work on it. This enables us to evaluate and demonstrate new features.

**Comment by ueisele** [comment]
I upgraded all Docker images as well as the Java libraries for Ksql and Kafka Connect to CP 6.0 (see branch _[ueisele-cp6](https://github.com/NovatecConsulting/showcase-kafka-iot-emob/tree/ueisele-cp6)_  of this repository).

All infrastructure services are able to start. However, unfortunately it seems that the Kafka Connector _mongodb/kafka-connect-mongodb_ is not compatible with CP 6.0. The deployment of the following connector configuration fails and on the server a _NullPointerException_ is thrown.

```
{
    "name": "mongodb__wallbox_charge__sink",
    "config": {
        "connector.class": "com.mongodb.kafka.connect.MongoSinkConnector",
        "tasks.max": 1,
        "topics": "wallbox_charge__sink",
        "connection.uri": "mongodb://mongo:27017",
        "database": "mongoDB",
        "collection": "WallboxCharge",
        "value.converter": "io.confluent.connect.avro.AvroConverter",
        "value.converter.schema.registry.url": "http://schema-registry:8081"
    }
}
```
I deployed the configuration file via the [Connect Rest API](https://docs.confluent.io/current/connect/references/restapi.html#post--connectors):
```
$ curl -s -X POST -H "Content-Type: application/json" -d@kafkaconnect/config/connect-mongodb-sink.json http://connect:8083/connectors
{"error_code":500,"message":null}
```
As response, `{"error_code":500,"message":null}` is returned and on the Kafka Connect server the following exception has been thrown:
```
[2020-10-16 16:58:53,558] INFO Cluster created with settings {hosts=[mongo:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms', maxWaitQueueSize=500} (org.mongodb.driver.cluster)
[2020-10-16 16:58:53,595] INFO Opened connection [connectionId{localValue:1, serverValue:20}] to mongo:27017 (org.mongodb.driver.connection)
[2020-10-16 16:58:53,608] INFO Monitor thread successfully connected to server with description ServerDescription{address=mongo:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, version=ServerVersion{versionList=[4, 4, 0]}, minWireVersion=0, maxWireVersion=9, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=2267081, setName='rs0', canonicalAddress=172.26.0.5:27017, hosts=[172.26.0.5:27017], passives=[], arbiters=[], primary='172.26.0.5:27017', tagSet=TagSet{[]}, electionId=7fffffff0000000000000001, setVersion=87063, lastWriteDate=Fri Oct 16 16:58:52 GMT 2020, lastUpdateTimeNanos=39675818171049} (org.mongodb.driver.cluster)
[2020-10-16 16:58:53,621] ERROR Uncaught exception in REST call to /connectors/mongodb__wallbox_charge__sink/config (org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper)
java.lang.NullPointerException
 	at org.apache.kafka.connect.runtime.WorkerConfigDecorator$MutableConfigInfos.lambda$removeAllWithName$0(WorkerConfigDecorator.java:295)
	at org.apache.kafka.connect.runtime.WorkerConfigDecorator$MutableConfigInfos.removeAll(WorkerConfigDecorator.java:305)
	at org.apache.kafka.connect.runtime.WorkerConfigDecorator$MutableConfigInfos.removeAllWithName(WorkerConfigDecorator.java:294)
 	at org.apache.kafka.connect.runtime.WorkerConfigDecorator$DecorationPattern.filterValidationResults(WorkerConfigDecorator.java:432)
	at org.apache.kafka.connect.runtime.WorkerConfigDecorator.lambda$decorateValidationResult$5(WorkerConfigDecorator.java:273)
 	at java.base/java.util.Collections$SingletonList.forEach(Collections.java:4856)
 	at org.apache.kafka.connect.runtime.WorkerConfigDecorator.decorateValidationResult(WorkerConfigDecorator.java:273)
	at org.apache.kafka.connect.runtime.AbstractHerder.validateConnectorConfig(AbstractHerder.java:392)
 	at org.apache.kafka.connect.runtime.AbstractHerder.lambda$validateConnectorConfig$1(AbstractHerder.java:326)
 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Other connectors could been successfully deployed. Therefore I assume an incompatibility is the cause.
I tried it with version _mongodb/kafka-connect-mongodb:1.2.0_ and _mongodb/kafka-connect-mongodb:1.3.0_.

You can find the current state of the upgrade in the branch _[ueisele-cp6](https://github.com/NovatecConsulting/showcase-kafka-iot-emob/tree/ueisele-cp6)_ of this repository.

**Comment by nadjahagen** [comment]
I experienced the same issue when trying to solve this. If an incompatibility is really the cause of this issue, there is no version of the MongoDB connector that tried to fix this. The Confluent Platform 6.0 was released on 24th September while the latest version of the MongoDB connector _mongodb/kafka-connect-mongodb:1.3.0_ was released on 30th September but only included an update of the Java driver and no other changes. In _mongodb/kafka-connect-mongodb:1.4.0-SNAPSHOT_ are only minor changes included which didn't seem to solve this as tested.
